### ISSUE TITLE
Error: from glmnet Fortran code (error code 7777); All used predictors have zero variance

### ISSUE THREAD

**Issue by MickMacq** [thomasp85/lime#3]
Hi!

Thanks for the great implementation of LIME in R. I think packages such as this will become extremely popular in the future. Maybe this question is best directed at the developer of glmnet...but I thought I would start here.

I am attempting to use LIME on one of my projects but continue to bump into the below error when I run the created 'explain' function after having run the lime() function:
"Error: from glmnet Fortran code (error code 7777); All used predictors have zero variance"

I haven't been able to figure out the reason for the above error - it just seems to pop up when I add additional features (or also when I add records) to my training data. All of the columns I use have some variance so I don't think the error is correct. Googling the error also hasn't helped so far. Note that I don't believe that the problem is with the test data since I can change that and the error persists.

Below is a (hopefully) reproducible example that shows how adding one extra feature (column) to the training data can trigger the error.

-----------------------

```r
# Load libraries
library(caret)
library(lime)

# Manually create example training data
col1 <- c(0.25, 0.25, 0.89, -1.65, 0.89, -0.38, -1.65, 0.89, -0.38, 0.89)
col2 <- c(-1.25, 0.99, 0.44, -0.28, -1.86, 0.82, 0.82, -0.28, 0.99, -0.28)
col3 <- c(-0.92, -0.92, -0.92, 0.97, 0.42, -0.92, -0.92, 1.31, 0.66, 1.23)
col4 <- c(0.43, 0.79, -2.77, -0.07, 0.41, 0.12, 0.11, 0.32, 0.32, 0.32)
col5 <- c(0.65, 0.51, -2.78, 0.28, 0.46, 0, -0.05, 0.34, 0.28, 0.3)
col6 <- c(0.17, 1.07, 0.17, 0.17, 0.39, -2.54, 0.17, 0.7, -0.7, 0.41)
col7 <- c(0.42,0.56, 0.09, 0.32, 0.38, 0.46, -0.09, -2.8, 0.31, 0.34)
col8 <- c(0.21,0.47, 0.21, 0.21, 0.36, -0.18, 0.21, 0.8, -2.75, 0.47)
col9 <- c(0.28, 0.28, 0.28, 0.28, -2.84, 0.28, 0.28, 0.44, 0.28, 0.46)

# Combine training data into data frames
data1Train <- data.frame(col1, col2, col3, col4, col5, col6, col7, col8) # LIME works with this data
data2Train <- data.frame(col1, col2, col3, col4, col5, col6, col7, col8, col9) # LIME doesn't work with this data

# Create training data target variable
targetTrain <- as.factor(c(1, 0, 1, 0, 1, 0, 0, 0, 1, 1))

# Manually create example test data
col1 <- c(4.69, 4.69)
col2 <- c(-0.59, -0.94)
col3 <- c(1.35, -0.92)
col4 <- c(-2.77, -0.7)
col5 <- c(-2.78, 0.21)
col6 <- c(-2.54, 0.02)
col7 <- c(-2.8, 0.07)
col8 <- c(-2.75, 0.18)
col9 <- c(-2.84, -2.84)

# Combine test data into data frames
data1Test <- data.frame(col1, col2, col3, col4, col5, col6, col7, col8)
data2Test <- data.frame(col1, col2, col3, col4, col5, col6, col7, col8, col9)

# Create test data target variable (note that this data is not used below)
targetTest <- as.factor(c(0, 1))

# Run first model and explain predictions using lime
model1       <- train(data1Train, targetTrain, method = 'rf', tuneLength = 1, trControl = trainControl(method = "none"))
explain1     <- lime(data1Train, model1)
explanation1 <- explain1(data1Test, n_labels = 1, n_features = 2)
plot_features(explanation1, ncol = 1)

# Run second model and explain predictions using lime
model2       <- train(data2Train, targetTrain, method = 'rf', tuneLength = 1, trControl = trainControl(method = "none"))
explain2     <- lime(data2Train, model2)
explanation2 <- explain2(data2Test, n_labels = 1, n_features = 2)
plot_features(explanation2, ncol = 1)
```

**Comment by thomasp85** [comment]
Thanks for this - I'll look into it. Next time do format example code as code so it is easier to run (I've done it for you this time).

**Comment by thomasp85** [comment]
I can reproduce. Will debug

**Comment by thomasp85** [comment]
Fixed in 02dcadd

The problem was that your test data was on the edge of the training data and you used binning of continuous variables. This resulted in a column of permutations that all were in the same bin. Zero-variance columns are now removed by default
